docs/mkdocs/docs/api/macros/json_diagnostic_positions.md

```diff
@@ -6,17 +6,20 @@
 
 This macro enables position diagnostics for generated JSON objects.
 
-When enabled, two new properties: `start_pos()` and `end_pos()` are added to `nlohmann::json` objects and fields. `start_pos()` returns the start position of
+When enabled, two new properties: `start_pos()` and `end_pos()` are added to `nlohmann::basic_json` objects and fields. `start_pos()` returns the start position of
 that JSON object/field in the original string the object was parsed from. Likewise, `end_pos()` returns the end position of that JSON object/field in the
 original string the object was parsed from.
 
-For objects and arrays, the start and end positions represent the positions of the opening and closing braces or brackets, respectively. For fields, the start
-and end positions represent either the opening and closing quotes for that field's value or the first and character after last in the field's numerical or
-predefined true/false/null values.
+For objects and arrays, the start position represents the position of the opening brace or bracket. The end position is the character after the closing
```

Comment:
One nit with the current version, we use a period between the start/end explanations for objects/arrays, and a comma for the fields version, should be consistent